### PR TITLE
Improve test coverage in utils

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Adds test coverage for jsonutils.
Removes unused and untested stringstack.

### CODE REVIEW

**Comment by obnoxxx** [comment]
@centos-ci ok to test

**Comment by phlogistonjohn** [comment]
Applied changes to check err value after http.NewRequest is called

File: pkg/utils/jsonutils_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		"failed check", dest.Bar, "==", 88)
+}
+
+func TestGetJsonFromRequestBadJson(t *testing.T) {
+	dest := &testDest{}
+	req, err := http.NewRequest(
+		http.MethodPut,
+		"/foo/bar",
+		bytes.NewBufferString(`{"foo": `))
+	tests.Assert(t, err == nil, "err is nil", err)
```
Comment (obnoxxx) [inline pkg/utils/jsonutils_test.go]: Er, now I read it again, isn't the logic reversed: The Assert will trigger when the err == nil condition is *not* met, hence, the error message should not read "err is nil" but rather reflect the fact that an error occurred. like
```
tests.Assert(t, err == nil, "Error calling http.NewRequest:", err)
```
I think several messages below have this same issue...

File: pkg/utils/jsonutils_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	resp := &http.Response{
+		Status:     "200 OK",
+		StatusCode: 200,
+		Body:       dummyCloser{bytes.NewBufferString(`{"foo": "port", "bar": 22}`)},
+	}
+	dest := &testDest{}
+	err := GetJsonFromResponse(resp, dest)
+	tests.Assert(t, err == nil, "err is nil", err)
+	tests.Assert(t, dest.Foo == "port", "failed check:", dest.Foo, "==", "port")
+	tests.Assert(t, dest.Bar == 22, "failed check", dest.Bar, "==", 22)
```
Comment (obnoxxx) [inline pkg/utils/jsonutils_test.go]: Isn't this kind of message somwhat awkward to read? How about
```
tests.Assert(t, dest.Bar == 22, "Expected Bar == 22, got ", dest.Bar)
```

Comment (phlogistonjohn) [reply pkg/utils/jsonutils_test.go]: Yes. These are better.

I have some ideas about reducing the boilerplate in the Assert calls, but I wanted to prototype them first, or look around to see if there are other Go libraries that do something similar to what I was thinking. Regardless of my grand plans, I will incorporate these suggestions.

**Comment by phlogistonjohn** [comment]
Applied changes to improve assert details.

**Comment by phlogistonjohn** [comment]
Hmm. centos-ci failed but I don't see how my changes caused the failure. I think it must have been caused by something else.

**Comment by obnoxxx** [comment]
no idea. seen it before. some issue in the machine maybe...

**Comment by obnoxxx** [comment]
@centos-ci retest please

**Comment by obnoxxx** [comment]
@centos-ci re-test this please

**Comment by phlogistonjohn** [comment]
@obnoxxx - ping. Any thoughts on this updated version?

**Comment by obnoxxx** [comment]
Will merge once the year is fixed, @phlogistonjohn .

**Comment by phlogistonjohn** [comment]
Oops. Copy-n-paste. Fixed now.

**Comment by obnoxxx** [comment]
Ignoring centos-ci failure since only the year in the comment changed. LGTM!

**Comment by phlogistonjohn** [comment]
Awesome! Thank you.

**Pull request merged**